src/Analytics/BasicProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Message\Mothership\CMS\Analytics;
+
+/**
+ * @author Samuel Trangmar-Keates
+ *
+ * A basic provider class which is constructed with the reference, params and names
+ * required. Useful for basic providers.
+ *
+ * If there is a more complecated provider, then AnalyticsProviderInterface should
+ * be implemented
+ */
+class BasicProvider implements AnalyticsProvidorInterface
+{
+	/**
+	 * @var string
+	 */
+	private $_name;
+
+	/**
+	 * @var string
+	 */
+	private $_viewRef;
+
+	/**
+	 * @var array
+	 */
+	private $_params;
+
+	/**
+	 * Construct the object with the correct fields.
+	 * 
+	 * @param string $name    The object identifier
+	 * @param string $viewRef The view reference
+	 * @param array  $params  The parameters to pass to the view
+	 */
+	public function __construct($name, $viewRef, $params = [])
```

Comment:
I don't know if it's overkill, but the other parts of MS that are designed like this, each type has its own class. I like the idea of this class but I still think that there should be a GoogleAnalyticsProvider class that extends this class. That way it's kind of easier to look in the directory and see which ones it supports